Docs: Add Troubleshooting for CONTRIBUTING.md Steps

## Application Errors
- [ ] Problem: JsonWebTokenError: invalid signature -- Solution: Clear the browser cookies for the _localhost_ domain, or use an incognito browser window to open the site.
- [ ] Problem: Where do application registration and other emails go? -- Solution: All email are captured and saved by Mailhog. Simply open http://localhost:8025/ to see captured emails. For account registration, the email validation link can be opened in the browser to complete the registration process.
- [ ]  The application shows _"Something went wrong relation "chapters" does not exist"_ - Solution:  The [database needs to be initialized](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md#initializing-the-database).
- [ ] The applications shows "_The table `public.chapters` does not exist in the current database."_ or _"Invalid 'prisma_1.prisma.chapters.findMany()"_ -- Solution: The database hasn't been created. Run `npm run db:reset` to clear and re-create the database tables.
 
## Linux Docker Troubleshooting
- [ ] `docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', PermissionError(13, 'Permission denied'))`
- [ ] Run `sudo docker-compose up` or add non-privileged user to the docker group as described in https://docs.docker.com/engine/install/linux-postinstall/ If Docker was run as sudo before adding your user to the docker group
- [ ] "WARNING: Error loading config file: /home/user/.docker/config.json " see https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user

## Node / NPM
- [ ] **Problem:** notsup Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"}) - **Solution:** Upgrade to npm version 7 or 8, like `npm install -g npm@latest` or by getting the latest version of NodeJS from your Linux package manager. Or, install [nvm](https://github.com/nvm-sh/nvm).
- [ ] **Problem:** Running _npm i_ gives the error "Module not found: Can't resolve 'framer-motion'" - **Solution:** Upgrade to npm version 7 or 8, like `npm install -g npm@latest` or by getting the latest version of NodeJS from your Linux package manager. Or, install [nvm](https://github.com/nvm-sh/nvm).
- [ ] **Problem:**
```npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, rename '/usr/lib/node_modules/npm'
 or 
Error: EACCES: permission denied, rename '/usr/bin/npm'
```
**Solution:** https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally or else https://stackoverflow.com/a/51024493
- [ ] If using Windows / WSL + Ubuntu Terminal, "npx /bin/sh^M bad interpreter: No such file or directory" likely happens when Node.js is already installed on Windows, but it also needs to be [installed within the Linux subsystem / terminal](https://nodejs.org/en/download/package-manager/), such as [installing it with _apt_ on in Ubuntu](https://nodejs.org/en/download/package-manager/#debian-and-ubuntu-based-linux-distributions).

## ElephantSQL
- _Error during schema drop: QueryFailedError: must be owner of view pg_stat_statements_ while running _npm run db:reset_ - Solution: the free tier of ElephantSQL doesn't allow concurrent connections, so it's necessary to run database commands one-at-a-time, like _npm run db:drop_ then _npm run db:sync_ and then _npm run db:seed_. Alternatively, a paid plan on ElephantSQL would avoid this issue. - via https://github.com/freeCodeCamp/chapter/issues/762

## General Docker Troubleshooting
- [Linux](https://docs.docker.com/engine/install/linux-postinstall/)
- [Windows](https://docs.docker.com/docker-for-windows/troubleshoot/)
- [Mac](https://docs.docker.com/docker-for-mac/troubleshoot/)
- [ ] Make sure Docker isn't already running and bound to port 54320 - ERROR: for chapter-npx-test_db_1  Cannot start service db: driver failed programming external connectivity on endpoint chapter-npx-test_db_1 (....): Bind for 0.0.0.0:54320 failed: port is already allocated. ERROR: for db  Cannot start service db: driver failed programming external connectivity on endpoint chapter-npx-test_db_1
- [ ] To shut down Docker on Linux, press Ctrl + C in the terminal where it was started.
- [ ] To purge old images, which can sometimes be the source of errors, run `docker-compose rm && docker image prune -a`


### WSL
- [ ] [General WSL troubleshooting](https://docs.microsoft.com/en-us/windows/wsl/install-win10#troubleshooting-installation)
- [ ] "WSL2 Installation is Incomplete" - Follow the steps to install the missing component
- [ ] WslRegisterDistribution failed with error: 0x80070057 - this is common on older CPUs which does not support Virtulization, HyperV or SLAT - Solution: Open Powershell and change back to WSL 1 using ``wsl --set-default-version 1``

### Windows Docker Troubleshooting
- [ ] "Docker Desktop requires Windows 10 Pro/Enterprise (######+) or Windows 10 Home (#####.####+)" - Windows 10 is running an old / unsupported version and needs to be updated before Docker will install.
- [ ] "Failed to execute script docker-composer"  - Solution: Start Docker Desktop via the Windows Start Menu and restart Git Bash
- [ ] Docker / WSL2 consumes all your system's RAM and is very slow - [Limit resources](https://medium.com/@lewwybogus/how-to-stop-wsl2-from-hogging-all-your-ram-with-docker-d7846b9c5b37)
- [ ] "Windows Defender Firewall has blocked some features of this app"  - Allow Access

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -621,7 +621,7 @@ If you see a JsonWebTokenError in the console, it's likely that you need to clea
 <details>
  <summary>Node / NPM TroubleShooting</summary>
 
-* **Problem:** You are getting this errors
+* **Problem:** You are getting this error
    > notsup Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"} </br>
 
   **Solution:** Upgrade to npm version 7 or 8, like `npm install -g npm@latest` or by getting the latest version of NodeJS from your Linux package manager. Or, install [nvm](https://github.com/nvm-sh/nvm).
@@ -640,3 +640,12 @@ If you see a JsonWebTokenError in the console, it's likely that you need to clea
 
   If your problem isn't included above. Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
 </details>
+
+<details>
+ <summary>ElephantSQL TroubleShooting</summary>
+
+* **Problem:** You are getting this error running `npm run db:reset`
+   > Error during schema drop: QueryFailedError: must be owner of view pg_stat_statements </br>
+
+  **Solution:** the free tier of ElephantSQL doesn't allow concurrent connections, so it's necessary to run database commands one-at-a-time, like `npm run db:drop` then `npm run db:sync` and then `npm run db:seed`. Alternatively, a paid plan on ElephantSQL would avoid this issue. [ElephantSQL Drop Schema error #762](https://github.com/freeCodeCamp/chapter/issues/762).
+  </details>

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -583,17 +583,12 @@ When not running locally, the client needs to be passed the server's location by
 * **Problem:**  Where do application registration and other emails go? </br>
   **Solution:** All email are captured and saved by Mailhog. Simply open [http://localhost:8025/](http://localhost:8025/) to see captured emails. For account registration, the email validation link can be opened in the browser to complete the registration process.
 
-* **Problem:** The application shows 
-  > Something went wrong relation "chapters" does not exist. </br>
-
-  **Solution:**  The [database needs to be initialized](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md#initializing-the-database).
-
 * **Problem:** The applications shows 
   > _The table `public.chapters` does not exist in the current database_</br>
   > or
   > *Invalid'prisma_1.prisma.chapters.findMany()* </br>
 
-  **Solution:** The database hasn't been created. Run `npm run db:reset` to clear and re-create the database tables.
+  **Solution:** The [database needs to be initialized](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md#initializing-the-database). Run `npm run db:reset` to clear and re-create the database tables.
 
   If your problem isn't included above. Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
 </details>

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -572,10 +572,6 @@ When not running locally, the client needs to be passed the server's location by
 
 # Troubleshooting
 
-Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
-
-If you see a JsonWebTokenError in the console, it's likely that you need to clear the token from your browser's local storage.
-
 <details>
  <summary>Application Errors</summary>
 

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -575,3 +575,4 @@ When not running locally, the client needs to be passed the server's location by
 Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
 
 If you see a JsonWebTokenError in the console, it's likely that you need to clear the token from your browser's local storage.
+<details><summary>Application Errors</summary>

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -575,7 +575,7 @@ When not running locally, the client needs to be passed the server's location by
 <details>
  <summary>Application Errors</summary>
 
-* **Problem:** 
+* **Problem:** The application shows
    > _JsonWebTokenError: invalid signature._ </br>
 
   **Solution:** Clear the browser cookies for the localhost domain, or use an incognito browser window to open the site.

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -575,4 +575,18 @@ When not running locally, the client needs to be passed the server's location by
 Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
 
 If you see a JsonWebTokenError in the console, it's likely that you need to clear the token from your browser's local storage.
+
 <details><summary>Application Errors</summary>
+
+  * **Problem:** JsonWebTokenError: invalid signature. </br>
+  **Solution:** Clear the browser cookies for the localhost domain, or use an incognito browser window to open the site.
+
+  * **Problem:**  Where do application registration and other emails go? </br>
+  **Solution:** All email are captured and saved by Mailhog. Simply open [http://localhost:8025/](http://localhost:8025/) to see captured emails. For account registration, the email validation link can be opened in the browser to complete the registration process.
+
+  * **Problem:** The application shows "Something went wrong relation "chapters" does not exist". </br>
+  **Solution:**  The [database needs to be initialized](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md#initializing-the-database).
+
+  * **Problem:** The applications shows "_The table `public.chapters` does not exist in the current database_" or "*Invalid'prisma_1.prisma.chapters.findMany()*" </br>
+  **Solution:** The database hasn't been created. Run `npm run db:reset` to clear and re-create the database tables.
+</details>

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -648,7 +648,9 @@ If you see a JsonWebTokenError in the console, it's likely that you need to clea
    > Error during schema drop: QueryFailedError: must be owner of view pg_stat_statements </br>
 
   **Solution:** the free tier of ElephantSQL doesn't allow concurrent connections, so it's necessary to run database commands one-at-a-time, like `npm run db:drop` then `npm run db:sync` and then `npm run db:seed`. Alternatively, a paid plan on ElephantSQL would avoid this issue. [ElephantSQL Drop Schema error #762](https://github.com/freeCodeCamp/chapter/issues/762).
-  </details>
+
+  If your problem isn't included above. Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
+</details>
 
 <details>
  <summary>General Docker TroubleShooting</summary>
@@ -659,13 +661,15 @@ If you see a JsonWebTokenError in the console, it's likely that you need to clea
   [Mac](https://docs.docker.com/desktop/mac/troubleshoot/)
 
 * **Problem:** You are getting this error running
-   >  ERROR: for chapter-npx-test_db_1 Cannot start service db: driver failed programming external connectivity on endpoint chapter-npx-test_db_1 (....): Bind for 0.0.0.0:54320 failed: port is already allocated. ERROR: for db Cannot start service db: driver failed programming external connectivity on endpoint chapter-npx-test_db_1</br>
+   > ERROR: for chapter-npx-test_db_1 Cannot start service db: driver failed programming external connectivity on endpoint chapter-npx-test_db_1 (....): Bind for 0.0.0.0:54320 failed: port is already allocated. ERROR: for db Cannot start service db: driver failed programming external connectivity on endpoint chapter-npx-test_db_1</br>
 
   **Solution:** Make sure Docker isn't already running and bound to port 54320.
 
 * **Note:** To shut down Docker on Linux, press Ctrl + C in the terminal where it was started.
 
 * **Note:** To purge old images, which can sometimes be the source of errors, run `docker-compose rm && docker image prune -a`
+
+  If your problem isn't included above. Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
 </details>
 
 <details>
@@ -675,12 +679,40 @@ If you see a JsonWebTokenError in the console, it's likely that you need to clea
   [WSL](https://docs.microsoft.com/en-us/windows/wsl/install#troubleshooting-installation)</br>
 
 * **Problem:** You are getting this error
-   >  WSL2 Installation is Incomplete</br>
+   > WSL2 Installation is Incomplete</br>
 
   **Solution:** Follow the steps to install the missing component.
 
 * **Problem:** You are getting this error
-   >   WslRegisterDistribution failed with error: 0x80070057</br>
+   > WslRegisterDistribution failed with error: 0x80070057</br>
 
   **Solution:** this is common on older CPUs which does not support Virtulization, HyperV or SLAT - Solution: Open Powershell and change back to WSL 1 using `wsl --set-default-version 1`
+
+  If your problem isn't included above. Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
+</details>
+
+<details>
+ <summary>Windows Docker TroubleShooting</summary>
+
+* **Problem:** You are getting this error
+   > Docker Desktop requires Windows 10 Pro/Enterprise (######+) or Windows 10 Home (#####.####+)</br>
+
+  **Solution:** Windows 10 is running an old / unsupported version and needs to be updated before Docker will install.
+
+* **Problem:** You are getting this error
+   > Failed to execute script docker-composer</br>
+
+  **Solution:** Start Docker Desktop via the Windows Start Menu and restart Git Bash.
+
+* **Problem:** Docker / WSL2 consumes all your system's RAM and is very slow</br>
+
+  **Solution:** [Limit resources](https://medium.com/@lewwybogus/how-to-stop-wsl2-from-hogging-all-your-ram-with-docker-d7846b9c5b37)
+
+* **Problem:** You are getting this error
+   > Windows Defender Firewall has blocked some features of this app</br>
+
+  **Solution:**  Allow Access.
+
+
+  If your problem isn't included above. Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
 </details>

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -666,4 +666,21 @@ If you see a JsonWebTokenError in the console, it's likely that you need to clea
 * **Note:** To shut down Docker on Linux, press Ctrl + C in the terminal where it was started.
 
 * **Note:** To purge old images, which can sometimes be the source of errors, run `docker-compose rm && docker image prune -a`
-  </details>
+</details>
+
+<details>
+ <summary>WSL TroubleShooting</summary>
+
+- Install Linux for Windows with WSL:</br>
+  [WSL](https://docs.microsoft.com/en-us/windows/wsl/install#troubleshooting-installation)</br>
+
+* **Problem:** You are getting this error
+   >  WSL2 Installation is Incomplete</br>
+
+  **Solution:** Follow the steps to install the missing component.
+
+* **Problem:** You are getting this error
+   >   WslRegisterDistribution failed with error: 0x80070057</br>
+
+  **Solution:** this is common on older CPUs which does not support Virtulization, HyperV or SLAT - Solution: Open Powershell and change back to WSL 1 using `wsl --set-default-version 1`
+</details>

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -605,15 +605,38 @@ If you see a JsonWebTokenError in the console, it's likely that you need to clea
 <details>
  <summary>Linux Docker Troubleshooting</summary>
 
-* **Problem:** the application shows
+* **Problem:** You are getting this error
    > docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', PermissionError(13, 'Permission denied')) </br>
 
   **Solution:** Run `sudo docker-compose up` or add non-privileged user to the docker group as described in [https://docs.docker.com/engine/install/linux-postinstall/](https://docs.docker.com/engine/install/linux-postinstall/) If Docker was run as sudo before adding your user to the docker group.
 
-* **Problem:** The application shows 
+* **Problem:** You are getting this error 
   > "WARNING: Error loading config file: /home/user/.docker/config.json "  </br>
 
   **Solution:**  see [https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user](https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user).
+
+  If your problem isn't included above. Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
+</details>
+
+<details>
+ <summary>Node / NPM TroubleShooting</summary>
+
+* **Problem:** You are getting this errors
+   > notsup Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"} </br>
+
+  **Solution:** Upgrade to npm version 7 or 8, like `npm install -g npm@latest` or by getting the latest version of NodeJS from your Linux package manager. Or, install [nvm](https://github.com/nvm-sh/nvm).
+
+* **Problem:** You are getting these errors
+  > npm ERR! Error: EACCES: permission denied, rename '/usr/lib/node_modules/npm'</br>
+  > or </br>
+  > Error: EACCES: permission denied, rename '/usr/bin/npm' </br>
+
+  **Solution:**  see [https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally) or [https://stackoverflow.com/a/51024493](https://stackoverflow.com/questions/48910876/error-eacces-permission-denied-access-usr-local-lib-node-modules/51024493#51024493).
+
+  * **Problem:** if you are using Windows / WSL + Ubuntu Terminal, and the application shows
+   > npx /bin/sh^M bad interpreter: No such file or directory </br>
+
+  **Solution:** likely happens when Node.js is already installed on Windows, but it also needs to be [installed within the Linux subsystem / terminal](https://nodejs.org/en/download/package-manager/), such as installing it with [apt on in Ubuntu](https://nodejs.org/en/download/package-manager/#debian-and-ubuntu-based-linux-distributions).
 
   If your problem isn't included above. Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
 </details>

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -671,8 +671,7 @@ When not running locally, the client needs to be passed the server's location by
 <details>
  <summary>WSL TroubleShooting</summary>
 
-- Install Linux for Windows with WSL:</br>
-  [WSL](https://docs.microsoft.com/en-us/windows/wsl/install#troubleshooting-installation)</br>
+- Install Linux for Windows with [WSL](https://docs.microsoft.com/en-us/windows/wsl/install#troubleshooting-installation).</br>
 
 * **Problem:** You are getting this error
    > WSL2 Installation is Incomplete</br>

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -578,15 +578,23 @@ If you see a JsonWebTokenError in the console, it's likely that you need to clea
 
 <details><summary>Application Errors</summary>
 
-  * **Problem:** JsonWebTokenError: invalid signature. </br>
+* **Problem:** JsonWebTokenError: invalid signature. </br>
   **Solution:** Clear the browser cookies for the localhost domain, or use an incognito browser window to open the site.
 
-  * **Problem:**  Where do application registration and other emails go? </br>
+* **Problem:**  Where do application registration and other emails go? </br>
   **Solution:** All email are captured and saved by Mailhog. Simply open [http://localhost:8025/](http://localhost:8025/) to see captured emails. For account registration, the email validation link can be opened in the browser to complete the registration process.
 
-  * **Problem:** The application shows "Something went wrong relation "chapters" does not exist". </br>
+* **Problem:** The application shows 
+  > Something went wrong relation "chapters" does not exist. </br>
+
   **Solution:**  The [database needs to be initialized](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md#initializing-the-database).
 
-  * **Problem:** The applications shows "_The table `public.chapters` does not exist in the current database_" or "*Invalid'prisma_1.prisma.chapters.findMany()*" </br>
+* **Problem:** The applications shows 
+  > _The table `public.chapters` does not exist in the current database_</br>
+  > or
+  > *Invalid'prisma_1.prisma.chapters.findMany()* </br>
+
   **Solution:** The database hasn't been created. Run `npm run db:reset` to clear and re-create the database tables.
+
+  If your problem isn't included above. Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
 </details>

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -610,7 +610,7 @@ When not running locally, the client needs to be passed the server's location by
 </details>
 
 <details>
- <summary>Node / NPM TroubleShooting</summary>
+ <summary>Node / NPM Troubleshooting</summary>
 
 * **Problem:** You are getting this error
    > notsup Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"} </br>
@@ -633,7 +633,7 @@ When not running locally, the client needs to be passed the server's location by
 </details>
 
 <details>
- <summary>ElephantSQL TroubleShooting</summary>
+ <summary>ElephantSQL Troubleshooting</summary>
 
 * **Problem:** You are getting this error running `npm run db:reset`
    > Error during schema drop: QueryFailedError: must be owner of view pg_stat_statements </br>
@@ -644,7 +644,7 @@ When not running locally, the client needs to be passed the server's location by
 </details>
 
 <details>
- <summary>General Docker TroubleShooting</summary>
+ <summary>General Docker Troubleshooting</summary>
 
 - Docker Installation in different operating system:</br>
   [Linux](https://docs.docker.com/engine/install/linux-postinstall/)</br>
@@ -664,7 +664,7 @@ When not running locally, the client needs to be passed the server's location by
 </details>
 
 <details>
- <summary>WSL TroubleShooting</summary>
+ <summary>WSL Troubleshooting</summary>
 
 - Install Linux for Windows with [WSL](https://docs.microsoft.com/en-us/windows/wsl/install#troubleshooting-installation).</br>
 
@@ -682,7 +682,7 @@ When not running locally, the client needs to be passed the server's location by
 </details>
 
 <details>
- <summary>Windows Docker TroubleShooting</summary>
+ <summary>Windows Docker Troubleshooting</summary>
 
 * **Problem:** You are getting this error
    > Docker Desktop requires Windows 10 Pro/Enterprise (######+) or Windows 10 Home (#####.####+)</br>

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -649,3 +649,21 @@ If you see a JsonWebTokenError in the console, it's likely that you need to clea
 
   **Solution:** the free tier of ElephantSQL doesn't allow concurrent connections, so it's necessary to run database commands one-at-a-time, like `npm run db:drop` then `npm run db:sync` and then `npm run db:seed`. Alternatively, a paid plan on ElephantSQL would avoid this issue. [ElephantSQL Drop Schema error #762](https://github.com/freeCodeCamp/chapter/issues/762).
   </details>
+
+<details>
+ <summary>General Docker TroubleShooting</summary>
+
+- Docker Installation in different operating system:</br>
+  [Linux](https://docs.docker.com/engine/install/linux-postinstall/)</br>
+  [Windows](https://docs.docker.com/desktop/windows/troubleshoot/)</br>
+  [Mac](https://docs.docker.com/desktop/mac/troubleshoot/)
+
+* **Problem:** You are getting this error running
+   >  ERROR: for chapter-npx-test_db_1 Cannot start service db: driver failed programming external connectivity on endpoint chapter-npx-test_db_1 (....): Bind for 0.0.0.0:54320 failed: port is already allocated. ERROR: for db Cannot start service db: driver failed programming external connectivity on endpoint chapter-npx-test_db_1</br>
+
+  **Solution:** Make sure Docker isn't already running and bound to port 54320.
+
+* **Note:** To shut down Docker on Linux, press Ctrl + C in the terminal where it was started.
+
+* **Note:** To purge old images, which can sometimes be the source of errors, run `docker-compose rm && docker image prune -a`
+  </details>

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -573,7 +573,7 @@ When not running locally, the client needs to be passed the server's location by
 # Troubleshooting
 
 <details>
- <summary>Application Errors</summary>
+ <summary>Application Troubleshooting</summary>
 
 * **Problem:** The application shows
    > _JsonWebTokenError: invalid signature._ </br>

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -576,9 +576,12 @@ Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. 
 
 If you see a JsonWebTokenError in the console, it's likely that you need to clear the token from your browser's local storage.
 
-<details><summary>Application Errors</summary>
+<details>
+ <summary>Application Errors</summary>
 
-* **Problem:** JsonWebTokenError: invalid signature. </br>
+* **Problem:** 
+   > _JsonWebTokenError: invalid signature._ </br>
+
   **Solution:** Clear the browser cookies for the localhost domain, or use an incognito browser window to open the site.
 
 * **Problem:**  Where do application registration and other emails go? </br>
@@ -595,6 +598,22 @@ If you see a JsonWebTokenError in the console, it's likely that you need to clea
   > *Invalid'prisma_1.prisma.chapters.findMany()* </br>
 
   **Solution:** The database hasn't been created. Run `npm run db:reset` to clear and re-create the database tables.
+
+  If your problem isn't included above. Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
+</details>
+
+<details>
+ <summary>Linux Docker Troubleshooting</summary>
+
+* **Problem:** the application shows
+   > docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', PermissionError(13, 'Permission denied')) </br>
+
+  **Solution:** Run `sudo docker-compose up` or add non-privileged user to the docker group as described in [https://docs.docker.com/engine/install/linux-postinstall/](https://docs.docker.com/engine/install/linux-postinstall/) If Docker was run as sudo before adding your user to the docker group.
+
+* **Problem:** The application shows 
+  > "WARNING: Error loading config file: /home/user/.docker/config.json "  </br>
+
+  **Solution:**  see [https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user](https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user).
 
   If your problem isn't included above. Visit our [chat](https://chat.freecodecamp.org/channel/chapter) for assistance. Or, [create an issue for new bugs or topics](https://github.com/freeCodeCamp/chapter/issues).
 </details>

